### PR TITLE
Server dev

### PR DESCRIPTION
**Pull request by nks5295**
Handler code is good to go:
- relies on marshal code to work
  Server should be fine
- now initializes gwkv_server struct
- now relies on handler for message parsing

### CODE REVIEW

**Comment by phil-lopreiato** [comment]
Tests failed, see https://travis-ci.org/gwAdvNet2015/gw-kv-store/builds/59294121

```
make[1]: Entering directory `/home/travis/build/gwAdvNet2015/gw-kv-store/server'
gcc -pthread -c server-kv.c server-kv-run.c
In file included from server-kv.c:14:0:
handle_operation.h:10:28: fatal error: ../lib/marshal.h: No such file or directory
compilation terminated.
make[1]: *** [build-srcs] Error 1
make[1]: Leaving directory `/home/travis/build/gwAdvNet2015/gw-kv-store/server'
make: *** [server] Error 2
```

Looks like you have to fix a header location

**Comment by amygara** [comment]
@nks5295 Good progress, Travis fails because you need to link in your handle_operation objects, see server/test/Makefile for example. Almost ready to merge.

File: server/server-kv.h [hunk 1]
```diff
@@ -1,27 +1,13 @@
...
-
 struct pool_list {
 	int fd;
 	struct pool_list *next;
 };
 
-//void parse_message(char *message, int len, struct operation *operation);
-//void process_operation(struct operation *operation, char *message, int *len);
+// This is the hashtable on the server
+struct gwkv_server *ht;
```
Comment (amygara) [inline server/server-kv.h:9]: Why is this here and not in server-kv.c? I may just be being a dumb right now, but my gut says it should be in the .c.
